### PR TITLE
Analysis view UI plugin

### LINKED ISSUE #1218: Operator view: Simplify analysis overview

### PR DESCRIPTION
**Pull request by netr0m**
## What does this pull request change?
- Changes Analysis view to be UI-plugin based
- Adds separate Analysis views (UI-plugins) for "Inspector" and "Operators"+ (operators + experts)
  - Simplified view for inspectors
  - Hide "schedule", "access control" for non-experts
- Adds a utility for fetching roles, which accounts for any impersonated roles
- Require `type` attr for entities
- Removes the "heading" below the header, as it does not provide any additional value
- Resolves deprecation warnings on `rowsMax` => `maxRows` (EDS)
- Moved a lot of duplicated / shared stuff to `@dmt/common`
- Cleanup

## Why is this pull request needed?
- To use UI plugins for Analysis view
- To allow for simplified "Inspector" views

## Issues related to this change
- Resolves #1218

### CODE REVIEW

File: web/packages/plugins/apps/analysis-platform/src/modules/Analysis/components/AnalysisCard.tsx [hunk 1]
```diff
@@ -148,21 +148,23 @@ const AnalysisCard = (props: AnalysisCardProps) => {
...
-              />
-              <Tooltip title={'Not implemented'}>
-                <Button style={{ width: 'max-content' }} disabled>
-                  Configure schedule
-                  <Icons name="time" title="time" />
-                </Button>
-              </Tooltip>
-            </>
-          )}
+          {hasExpertRole(tokenData) &&
```
Comment (KristianKjerstad) [inline web/packages/plugins/apps/analysis-platform/src/modules/Analysis/components/AnalysisCard.tsx:151]: looks like hasExpertRole(tokenData) is present on all components inside <Card.Actions>. Is it possible to move the hasExpertRole to conditionally render the entire <Card.Actions> block?

**Review by KristianKjerstad — COMMENTED** [review]
nice!

I got an error when having the role "expert-operator" in the expert view. For other roles, the expert view works

image

File: web/packages/plugins/apps/analysis-platform/src/modules/Analysis/components/AnalysisCard.tsx (hunk 1 above)
Comment (netr0m) [reply web/packages/plugins/apps/analysis-platform/src/modules/Analysis/components/AnalysisCard.tsx:151]: Indeed! Fixed!

**Comment by netr0m** [comment]
> nice!
> 
> I got an error when having the role "expert-operator" in the expert view. For other roles, the expert view works
> 
> image

Fixed!

File: web/packages/plugins/apps/analysis-platform/src/modules/Analysis/AnalysisOverview.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { AuthContext, DmtSettings } from '@dmt/common'
 import React, { ReactNode, useContext } from 'react'
 import AnalysisTable from './components/AnalysisTable'
 import { Link, useLocation } from 'react-router-dom'
-import { hasExpertRole } from '../../utils/auth'
+import { hasExpertRole } from '@dmt/analysis-platform'
```
Comment (soofstad) [inline web/packages/plugins/apps/analysis-platform/src/modules/Analysis/AnalysisOverview.tsx:5]: Should a package not use relative references for internal imports?
Whats the pros/cons of doing it this way?

File: web/packages/plugins/ui/analysis/src/ExpertView.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from 'react'
+import { DmtUIPlugin, UIPluginSelector } from '@dmt/common'
+import { AnalysisInfoCard, AnalysisJobTable } from '@dmt/analysis-platform'
```
Comment (soofstad) [inline web/packages/plugins/ui/analysis/src/ExpertView.tsx:3]: These components, that are only used by these plugins, should be in this plugin package.
There should not be dependencies from plugins to any other "custom package" that "dmt/common"

**Review by soofstad — CHANGES_REQUESTED** [review]

File: api/src/home/analysis-platform/job_handlers/azure_container_instances/__init__.py [hunk 4]
```diff
@@ -157,13 +157,13 @@ def progress(self) -> Tuple[JobStatus, str]:
 
         job_status = self.job.status
 
-        match (status, exit_code):
-            case ("Running", None):
+        match (status, exit_code):  # noqa: E211,E999
```
Comment (soofstad) [inline api/src/home/analysis-platform/job_handlers/azure_container_instances/__init__.py:160]: Is this still needed even after new venv?

**Review by soofstad — APPROVED** [review]

**Pull request merged**